General Definition Cleanup - 09/03/2024

robertgk2017 opened the pull request:
TES4\ GRAS - 3x Unknowns confirmed Unused by Cobb
TES4 \ WATR - Set default values + Updated Required tag
TES4 \ WEAP - Small tidy-up

FO3/FNV \ GRAS - 3x Unknowns confirmed Unused by Cobb
FO3/FNV \ QUST \ DATA - Unknown 4 = "Default Script Processing Delay"
FO3/FNV \ REFR \ XPRM - Updated Unknowns to actually be Unknown

TES5 \ ADDN \ DNAM \ Flags - Uknown 1 = "Master Particle System"
TES5 \ EXPL - Added EAMT, confirmed by Cobb to be present in the loader
TES5 \ FLOR - Decoded FNAM, Cobb confirmed same as ACTI \ FNAM
TES5 \ GRAS - 3x Unknowns confirmed Unused by Cobb
TES5 \ IMGS - Finished decoding ENAM + Tidy-up
TES5 \ INFO - Decoded DATA Sub-Record
TES5 \ MATO - Simplified DATA Sub-record
TES5 \ NPC_ \ NAM9 - Last unknown float is VampireMorph, confirmed by Cobb
TES5 \ SCEN \ DNAM - Fixed Typo
TES5 \ SHOU - Added ETYP, confirmed by Cobb to be present

FO4 \ GRAS - 3x Unknowns confirmed Unused by Cobb
FO4 \ NOTE - Cleaned up definition
FO4 \ ZOOM - Tidy up, set wbZoomOverlayEnum to Common

FO76 \ NOTE - Cleaned up definition
FO76 \ PERK \ DATA - Resolves crash
FO76 \ ZOOM - Tidy up, set wbZoomOverlayEnum to Common

SF1 \ ACHR - Renamed XEZN from Encounter Location to Location Override to match CK
SF1 \ ACTI - Moved WTFM to correct position
SF1 \ AFFE - Added new PNAM sub-record
SF1 \ AMDL - Added BaseFormComponents, New Flag, and Updated field names
SF1 \ BIOM - Updated GNAM allowed form id signatures
SF1 \ CELL - Removed XEZN as it doesn't actually exist for Cells
SF1 \ INGR - Decoded Ingredient Record
SF1 \ IPDS - Added new ENAM sub-record
SF1 \ NOTE - Decoded Note Record
SF1 \ NPC_ - Updated ONA2 flags
SF1 \ PNDT \ EOVR - Decoded new Sub-Record
SF1 \ REFR - Added XLIB Sub-Record
SF1 \ REFR - Renamed XEZN from Encounter Location to Location Override to match CK
SF1 \ REFR - Water Current Sub-Records Decoded
SF1 \ RSPJ - Updated Names to match the CK
SF1 \ RSPJ \ FVPA - Updated allowed signatures.
SF1 \ SCCO - Enabled SCCO Main Record
SF1 \ SCEN - Moved DEVT to correct position
SF1 \ TES4- Update File Flags
SF1 \ XTV2 - Added dfNotAlignable flag
SF1 \ WEAP - Added wbXALG
SF1 \ WKMF - Cleanup definition and move functions to Common
SF1 \ WRLD - Removed XEZN as it doesn't actually exist for Worldspaces
SF1 \ WTHR - WSLS/WSLD Decoded
SF1 \ ZOOM - Tidy up, Set wbZoomOverlayEnum to Common

ALL \ NAVI/NAVM - Refactored Definitions across the games, utilizing summaries/refined def methods.   Moved all navmesh related functions/variables to Common.  And simplified/collapsed unecessary split-up variables

ALL \ NOTE \ wbNoteTypeDecider - Moved to Common
ALL \ WRLD - Updated Required Tags and Default Values
ALL \ WRLD \ MHDT - Set to cpBenign as it is not a Rule of One sub-record and added SetSummaryKeys
ALL \ WRLD - Moved the .IncludeFlags for the various Large sub-records over to their respective common defs
ALL \ WRLD - Added IsRemovable callbacks to the sub-records that are toggled by the World AfterLoad/AfterSet
ALL \ WTHR - Updated default values and set Required where applicable based on output from every CK

TES4 - Simplify unneeded after sets
TES5 - Convert Boolean Ints to wbBoolEnum
TES5 - Various Required tags set based on CK
FO76 - Updated Default HEDR Version
FO76/SF1 - Update Default Form Version
SF1 - Add .ccc file support
SF1 - Update DLC list to reflect behavior in 1.14.70
SF1 - wbAddGroupOrders updated
SF1 - wbSoundReference Updated to fix 4 byte buffer
ALL - Remove unused variables
ALL - updated wbFromVersion + wbBelowVersion to not show the 'unused' field name when comparing many records
ALL - Updated wbQuestEventEnum implementation
ALL - Updated Record Flags implementation

Moved the following EnumDefs to Common
wbAggressionEnum
wbAlignmentEnum
wbArchtypeEnum
wbAssistanceEnum
wbAttackAnimationEnum
wbBlendModeEnum
wbBlendOpEnum
wbBodyLocationEnum 
wbConfidenceEnum 
wbCRCValuesEnum
wbMusicEnum
wbZTestFuncEnum

BSArchPro - Fix files in Music folder being erroneously moved to Sound + Compressed Check

xEdit - dfArrayCanBeEmpty - Fix Implementation
xEdit - dfArrayStaticSize - Added new flag to force an Array to a specific size and prevent adding/removing entries
xEdit - dfFlags - Sorted list of flags better
xEdit - SetIsRemovable - Enabled using this Set on Def Types outside ValueDef
xEdit - wbDontDrawColorText - Added boolean for toggling display of Color Field names in Summaries
xEdit - wbWorldAfterLoad - Combined related functions into one procedure and moved to Common
xEdit - wbWorldAfterSet - Added AfterSet to update sub-records based on the presence of a Parent Worldspace + Inherit flags.

Misc - Corrected spelling of Removeable to Removable across all files.

Updated README.md instructions for setting up Delphi.
Updated version to 4.1.5k

robertgk2017 committed:
WRLD \ MHDT - set to cpBenign as its not Rule of One

And setup Summary Keys for it, as well as moved the .IncludeFlags from the WRLD records into the common defs.

robertgk2017 committed:
Update README, Update Version to 4.1.5j, Add wbDontDrawColorText to wbInterface

robertgk2017 committed:
Re-Organized Common File, added Summary Keys to several WRLD/WTHR sub-records.

robertgk2017 committed:
Update README.md

robertgk2017 committed:
wbWorldAfterLoad condensed into 1 procedure and moved to Common

robertgk2017 committed:
SF1 \ RSPJ \ FVPA - Updated allowed signatures.

robertgk2017 committed:
SF1 \ RSPJ - Updated Names to match the CK

robertgk2017 committed:
FO76 \ PERK \ DATA - Resolves crash

robertgk2017 committed:
Unneeded dfFlag.

robertgk2017 committed:
Condensed/Simplified GameMode IfThens

robertgk2017 committed:
wbKeywrods moved to common

robertgk2017 committed:
SF1 - Add .ccc file support

robertgk2017 committed:
SF1 \ TES4 - Update File Flags

robertgk2017 committed:
SF1 \ XTV2 - Add dfNotAlignable flag

robertgk2017 committed:
Trim the WRLD AfterLoad tags

robertgk2017 committed:
Fix dfArrayCanBeEmpty implementation

robertgk2017 committed:
ALL \ WRLD - Updated Required Tags and Default Values

robertgk2017 committed:
SF1 \ WRLD \ NAM7 - Fix typo in name

robertgk2017 committed:
wbWorldAfterSet and spacing cleanup

robertgk2017 committed:
FO3/FNV \ REFR \ XPRM - Updated Unknowns to actually be Unknown

robertgk2017 committed:
FO76/SF1 - Update Default Form Version

robertgk2017 committed:
FO76 - Update HEDR Version

robertgk2017 committed:
TES5 \ SCEN \ DNAM - Fixed Typo

robertgk2017 committed:
xEdit - Add dfArrayStaticSize flag and sort dfFlags

robertgk2017 committed:
SF1 \ REFR \ Water Current Sub-Records Decoded

robertgk2017 committed:
SF1 - Increment Default Form Version to 576

robertgk2017 committed:
Not just yet on Starfield Form Version 576, ck doesn't support it yet.

robertgk2017 committed:
SF1 \ WTHR - WSLS/WSLD Decoded

robertgk2017 committed:
ALL \ WTHR - Updated default values and set Required where applicable from the CKs

robertgk2017 committed:
Fix TES5 \ WRLD \ PNAM not being Required

robertgk2017 committed:
Enable .SetIsRemovable on Def Types outside ValueDef

robertgk2017 committed:
Move records beginning with W to the bottom, sorted by name.

robertgk2017 committed:
2 more records to move

robertgk2017 committed:
SF1 \ PNDT \ EOVR - Decoded new sub-record

robertgk2017 committed:
FO4/FO76/SF1 \ ZOOM - Tidy up

robertgk2017 committed:
SF1 - wbAddGroupOrders updated

robertgk2017 committed:
SF1 \ INGR - Decoded Ingredient Record

robertgk2017 committed:
ALL \ NOTE \ wbNoteTypeDecider - Moved to Common

robertgk2017 committed:
FO4/FO76 \ NOTE - Cleaned up definition

robertgk2017 committed:
SF1 \ NOTE - Decoded Note Record

robertgk2017 committed:
FO4/FO76/SF1 \ ZOOM - Set wbZoomOverlayEnum to Common

robertgk2017 committed:
SF1 \ ZOOM - Update field names to better match CK

robertgk2017 committed:
SF1 \ WKMF - Cleanup definition and move functions to Common.

robertgk2017 committed:
SF1 \ AMDL - Added BaseFormComponents, New Flag, and Updated field names

robertgk2017 committed:
WTHR \ PNAM - Adds dfFastAssign to avoid an Assert

robertgk2017 committed:
ALL - Fix Worldspace Sub-Record IsRemovable Logic

robertgk2017 committed:
ALL - Quicky Tidy of Common variable list

robertgk2017 committed:
ALL - Add wbEnchantment common def

robertgk2017 committed:
TES5 - Convert Boolean Ints to wbBoolEnum

robertgk2017 committed:
ALL - Move wbCRCValuesEnum to Common

robertgk2017 committed:
ALL - Move wbBlendModeEnum to Common

robertgk2017 committed:
ALL - Move wbBlendOpEnum to Common

robertgk2017 committed:
ALL - Updated wbQuestEventEnum implementation

Added game mode IfThens so events only show up in games they exist in.

robertgk2017 committed:
TES4 \ WATR - Set default values + Updated Required tag

robertgk2017 committed:
TES4 \ WEAP - Small tidy-up

robertgk2017 committed:
TES5 \ IMGS - Finished decoding ENAM + Tidy-up

robertgk2017 committed:
TES5 \ MATO - Simplified DATA Sub-record

robertgk2017 committed:
TES5 \ INFO - Decoded DATA Sub-Record

robertgk2017 committed:
TES5 - Various Required tags set based on CK

robertgk2017 committed:
All - updated wbFromVersion + wbBelowVersion to not show the 'unused' field name when comparing many records

robertgk2017 committed:
ALL - Remove unused variables

Not even commented out uses of them, literally useless.

robertgk2017 committed:
ALL - Move wbMusicEnum to Common

robertgk2017 committed:
ALL - Move wbZTestFuncEnum to Common

robertgk2017 committed:
ALL - Move wbAggressionEnum to Common

LordNyriox commented:
@robertgk2017: 

Just so you know, there is a typo you forwarded when moving `wbZTestFuncEnum`: 

> 'Greater than or equal **than**'

…should be… 

> 'Greater than or equal **to**'

robertgk2017 commented:
OOPS, good catch!

robertgk2017 committed:
Typo Fix

robertgk2017 committed:
ALL - Move wbAlignmentEnum to Common

robertgk2017 committed:
ALL - Move wbArchtypeEnum to Common

robertgk2017 committed:
ALL - Move wbAssistanceEnum to Common

robertgk2017 committed:
Another Typo Fix

robertgk2017 committed:
ALL - Move wbAttackAnimationEnum to Common

robertgk2017 committed:
Last of the wbBoolEnums

robertgk2017 committed:
ALL - Move wbConfidenceEnum to Common

robertgk2017 committed:
even more wbBoolEnums

robertgk2017 committed:
TES4 - Simplify unneeded after sets

robertgk2017 committed:
Move wbBodyLocationEnum to Common

robertgk2017 committed:
SF1 - Update form version to 576

robertgk2017 committed:
SF1 - Shattered Space Update Part 1

wbBaseFormComponents \ BGSAnimationGraph_Component - Added dfAllowAnyMember flag as the sub-records are actually optional.

ACTI \ WTFM - Moved to correct order position

AFFE - Decoded new PNAM Sub-Record

IPDS - Decoded new ENAM Sub-Record

SCEN \ DEVT - Moved to correct order position

NPC_ \ ONA2 - Added Unknowns to new flags, needs further work as several of the structs changed lengths/behavior

REFR - Added XLIB sub-record, it's the same as previous games

SCCO - Enabled Scene Collections as they still exist, just were not used by Bethesda until now.

WEAP - Added wbXALG

robertgk2017 committed:
SF1 \ wbSoundReference - moved the 4 byte unknown padding.

robertgk2017 committed:
SF1 \ BIOM \ GNAM - Update allowed record references.

robertgk2017 committed:
SF1 - Update DLC List

robertgk2017 committed:
SF1 - Enable EYES record, it does actually exist.

robertgk2017 committed:
SF1 \ WATR - Update Record Flags

robertgk2017 committed:
SF1 \ TRAV - Add wbBaseFormComponents

robertgk2017 committed:
SF1 \ STDT - Add SNAM sub-receord + remove unneeded IncludeFlag

robertgk2017 committed:
SF1 \ PNDT - Additional Decoding

robertgk2017 committed:
SF1 \ BIOM - Update names + allowed referenced signatures

robertgk2017 committed:
SF1 - wbTerminalBackgroundEnum - updated name/entries

robertgk2017 committed:
SF1 \ PKIN - Added new unknown used flags, still need to grab names from CK.

robertgk2017 committed:
SF1 \ REFR - New Map Marker Types and Flag

robertgk2017 committed:
SF1 \ NPC - Update RaceSizeEnum and ONA2 flags.

robertgk2017 committed:
SF1 \ RACE - New Data flag

robertgk2017 committed:
SF1 \ PROJ - New Record and Data flags

robertgk2017 committed:
SF1 \ AFFE - New PNAM sub-record

robertgk2017 committed:
SF1 \ DIAL - New subtypes related to Vehicles

robertgk2017 committed:
SF1 \ CELL - Updated Data Flags

robertgk2017 committed:
SF1 \ BaseFormComponents \ BGSStoredTraversalsData - Updated allowed reference signatures

robertgk2017 committed:
SF1 \ DIAL - Sorted and added new SubTypes to wbDialogueSubtypeEnum

robertgk2017 committed:
SF1 \ wbPTDA - Added Unknown 9 and corresponding Union option

robertgk2017 committed:
SF1 \ wbLightAreaLightData - Added new Unknown Light Types

jbostrus committed:
SF1: CELL update unknowns

jbostrus committed:
SF1: PACK\PTDA\Type update unknowns

jbostrus committed:
SF1: DIAL - Subtype update unknowns

jbostrus committed:
SF1: PROJ update unknowns

robertgk2017 committed:
Merge pull request #1 from jbostrus/robert-misc

Update Unknowns

robertgk2017 committed:
SF1 \ TERM - Fully decoded every flag/subrecord

robertgk2017 committed:
SF1 \ TERM - Update Required Tags

robertgk2017 committed:
SF1 \ TERM - One more default

robertgk2017 committed:
SF1 \ TERM - Added DEFL subrecord

robertgk2017 committed:
SF1 \ TERM - Add another flag and more sub-records that CK allows

robertgk2017 committed:
Enable proper FlagDontShow behavior

robertgk2017 committed:
ALL - Cleanup implementation for Record Flags across all games

robertgk2017 committed:
ALL - Setup navmesh flag dontshows for various base object records

robertgk2017 committed:
TES4/FO3/FNV/TES5/FO4 \ GRAS - 3x Unknowns confirmed Unused by Cobb

robertgk2017 committed:
ALL \ NAVM/NAVI - Refactor Definition across all games

Implemented Summaries/Newer Simpler Definition Methods across the games.
Collapsed unecessary split variables
Moved all navmesh related functions to Common

robertgk2017 committed:
NAVI - Remove dfUnionStaticResolve

robertgk2017 committed:
NAVI - Add Island Union Name

robertgk2017 committed:
BSArchPro - Fix files in Music folder being erroneously moved to Sound + Compressed Check

robertgk2017 committed:
FO3/FNV \ QUST \ DATA - Unknown 4 = "Default Script Processing Delay"

robertgk2017 committed:
TES5 \ ADDN \ DNAM \ Flags - Uknown 1 = "Master Particle System"

robertgk2017 committed:
TES5 \ EXPL - Added EAMT, confirmed by Cobb to be present

robertgk2017 committed:
TES5 \ SHOU - Added ETYP, confirmed by Cobb to be present

robertgk2017 committed:
TES5 \ FLOR - Decoded FNAM, Cobb confirmed same as ACTI \ FNAM

robertgk2017 committed:
TES5 \ NPC_ \ NAM9 - Last unknown float is VampireMorph, confirmed by Cobb.

robertgk2017 committed:
Increment Version to 4.1.5k

robertgk2017 committed:
TES5/FO4/FO76/SF1 \ NAVI - Remove dfExcludeFromBuildRef, causes issues when trying to merge plugins editing navmesh

robertgk2017 committed:
TES5 \ WEAP - Add EFSD Sub-Record

robertgk2017 committed:
TES5 \ WEAP \ DNAM - Unknown field decoding from Cobb
